### DISCUSSION TITLE
APS ACK on Battery device ( not Receive on Idle, but doing polling)

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy-znp#197] [Q&A]
Guys I'm struggling to get sending messages to End devices like eTRV , which are not listening on idle but instead are doing polling.

This issue happen only with ZNP, everything works well on bellows. I feel the issue is related to https://github.com/Koenkk/Z-Stack-firmware/blob/3406c6a0cd2b755f302f4cacd07b37497a86cc08/coordinator/Z-Stack_3.x.0/firmware.patch#L349 where the APS ACK time as been reduced from 6s to 1s and the coordinator doesn't buffer enough the command to get it polled when a data request is coming from the end device

Please do note that I don't say that it never work, but it is more a ratio of 50/50, when the request is send in the right time.

I also wonder as ZNP is also trying different way to handle the failure at each attempt is that is also not jeopardising the chance to get it received by the end device. For instance ZNP is trying to send with IEEE, doing a route discovery, while there is no need and it should just wait for the polling happening

I wonder how is that managed in the ZHA implementation to understand if you have the same issue or if you do something special.

Would really appreciate your view

cc: https://github.com/zigbeefordomoticz/Domoticz-Zigbee/issues/1341

**Comment by MattWestb** [comment]
Some interesting reading how the Zigbee specification of pulling https://community.silabs.com/s/article/how-often-should-my-end-device-poll-its-parent-x?language=en_US.

So `ZigBee Pro PICS` is saying the the parent shall holding the message to its children for 7.680 seconds and the patch is setting 1 seconds for APS acks = you is likely loosing acks for APS commands from direct children.

Z-Stack is not Zigbee PRO / Zigbee 3 compatible at all.

**Comment by MattWestb** [comment]
One very interesting setting:
https://github.com/Koenkk/Z-Stack-firmware/blob/3406c6a0cd2b755f302f4cacd07b37497a86cc08/coordinator/Z-Stack_3.x.0/firmware.patch#L323-L325

The underlying IEEE 802.15.4 network is limiting broadcast in all routers to 9 / 8 seconds = 1.2 / seconds.
Making the coordinator sending near 10 times broadcast is only making the network blocking broadcast for not getting broadcast storms in the network. Is not Zigbee compliant and is not working in real life then all router is blocking  it = the network is blocked.
